src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -437,14 +436,11 @@ private static List<DataRange> dataRanges(ReadCommand command)
     {
         if (command instanceof SinglePartitionReadCommand)
         {
-            SinglePartitionReadCommand cmd = (SinglePartitionReadCommand) command;
-            DecoratedKey key = cmd.partitionKey();
-            return Lists.newArrayList(new DataRange(new Range<>(key, key), cmd.clusteringIndexFilter()));
+            return Lists.newArrayList(command.dataRange());
```

Comment:
Why is this wrapped in a list if it can be only one? Seems like it only ever is one.

Reply:
tl;dr It was part of [CASSANDRA-18480](https://issues.apache.org/jira/browse/CASSANDRA-18480) which is only implemented in the DataStax public fork right now, IIRC.